File: Java/Conditional Statements/Exercise/04. Toy Shop/ToyShop.java
import java.util.Scanner;

public class ToyShop {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        double puzzlePrice = 2.60, talkingDollPrice = 3.00, teddyBearPrice = 4.10, minionPrice = 8.20, toyTruckPrice = 2.00;

        double tourCost = Double.parseDouble(input.nextLine());
        int puzzles = Integer.parseInt(input.nextLine());
        int dolls = Integer.parseInt(input.nextLine());
        int teddyBears = Integer.parseInt(input.nextLine());
        int minions = Integer.parseInt(input.nextLine());
        int trucks = Integer.parseInt(input.nextLine());

        int orderedToys = puzzles + dolls + teddyBears + minions + trucks;

        double totalEarnings = (puzzles * puzzlePrice) + (dolls * talkingDollPrice) + (teddyBears * teddyBearPrice) + (minions * minionPrice) + (trucks * toyTruckPrice);

        //discount
        if(orderedToys >= 50){
            totalEarnings -= totalEarnings * 0.25;
        }

        //rent expenses
        totalEarnings -= totalEarnings * 0.1;

        if (totalEarnings >= tourCost) {
            System.out.printf("Yes! %.2f lv left.", totalEarnings - tourCost);
        } else {
            System.out.printf("Not enough money! %.2f lv needed.", tourCost - totalEarnings);
        }
    }
}


/*
⦁	Пъзел - 2.60 лв.
        ⦁	Говореща кукла - 3 лв.
        ⦁	Плюшено мече - 4.10 лв.
        ⦁	Миньон - 8.20 лв.
        ⦁	Камионче - 2 лв.
*/



/*
⦁	Цена на екскурзията - реално число в интервала [1.00 … 10000.00]
        ⦁	Брой пъзели - цяло число в интервала [0… 1000]
        ⦁	Брой говорещи кукли - цяло число в интервала [0 … 1000]
        ⦁	Брой плюшени мечета - цяло число в интервала [0 … 1000]
        ⦁	Брой миньони - цяло число в интервала [0 … 1000]
        ⦁	Брой камиончета - цяло число в интервала [0 … 1000]
*/


//Example input
/*
40.8
20
25
30
50
10
*/

File: Python/Basics/Conditional Statements/Exercises/ToyShop.py
earnings = 0
sale = False
saleAmount = 0 #Percentage
rentCost = 10 # Percentage
holidayCost = float(input())

jigsaws = int(input())
talkingDolls = int(input())
teddyBears = int(input())
minions = int(input())
trucks = int(input())

jigsawP = 2.60
dollsP = 3.00
bearP = 4.10
minionP = 8.20
truckP = 2.00

if jigsaws + talkingDolls + teddyBears + minions + trucks >= 50:
    sale = True
    saleAmount = 25

earnings += (jigsaws * jigsawP) + (talkingDolls * dollsP) + (teddyBears * bearP) + (minions * minionP) + (trucks * truckP)

if sale == True:
    earnings -= earnings * (saleAmount / 100)

earnings -= earnings * (rentCost / 100)

if earnings >= holidayCost:
    print(f"Yes! {earnings - holidayCost:.2f} lv left.")
else:
    print(f"Not enough money! {holidayCost - earnings:.2f} lv needed.")